### PR TITLE
Switch to using equip structures

### PR DESCRIPTION
**Pull request by shadowhand**
Refs equip/structure@9bb7222d9570

~~Since the method names have changed, this will need to be released in v2.x.~~ Added a compatibility trait to ensure backwards compatibility.

### CODE REVIEW

**Comment by shadowhand** [comment]
New structure package also need dosc!

**Comment by elazar** [comment]
@shadowhand I'd include the docs in this PR.

**Comment by shadowhand** [comment]
@elazar agreed!

**Comment by shadowhand** [comment]
Should I add a BC methods so that we can release this in v1.x? All it takes it adding a `withData -> withValues` alias.

**Comment by elazar** [comment]
@shadowhand I would, just to ease the transition, and mark them `@deprecated`.

File: src/Compatibility/StructureWithDataAlias.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Backwards compatability with Destrukt.
+     *
+     * @deprecated since 1.3.0, to be removed in 2.0.0
+     *
+     * @param array $data
+     *
+     * @return static
+     */
+    public function withData(array $data)
```
Comment (shadowhand) [inline src/Compatibility/StructureWithDataAlias.php]: @elazar compat added here.

File: tests/DirectoryTest.php [hunk 2]
```diff
@@ -20,7 +20,7 @@ protected function setUp()
 
     public function testDictionary()
     {
-        $this->assertInstanceOf('Destrukt\Dictionary', $this->directory);
+        $this->assertInstanceOf('Equip\Structure\Dictionary', $this->directory);
```
Comment (shadowhand) [inline tests/DirectoryTest.php]: Reverted all other tests to ensure that backwards compatibility is maintained.

**Comment by elazar** [comment]
:+1:

**Pull request merged**